src/views/Register.jsx

```diff
@@ -0,0 +1,107 @@
+import axios from 'axios';
+import { useState } from 'react';
+import { Link, useNavigate } from 'react-router-dom';
+import { useAuth } from '../contexts/auth-context';
+
+export default function Register() {
+  // Hooks
+  const [form, setFormData] = useState({ name: '', email: '', password: '' });
+  const [showPassword, setShowPassword] = useState(false);
+  const navigate = useNavigate();
+  const { login } = useAuth();
+
+  const { name, email, password } = form;
+  // Handlers
+  const formHandler = (field, value) => {
+    setFormData({ ...form, [field]: value });
+  };
+  const toggleCheckbox = () => setShowPassword((value) => !value);
+
+  const signUpHandler = async (e) => {
+    e.preventDefault();
+
+    try {
+      const response = await axios.post('/api/auth/signup', {
+        name,
+        email,
+        password,
+      });
+
+      const { encodedToken, createdUser } = response.data;
+      login({ token: encodedToken, user: createdUser });
+
+      navigate('/videos', { replace: true });
+    } catch (error) {
+      console.log(error);
```

Comment:
You could console your errors as console.error(error), that way your errors wou;d be separated from your console.log() and help in debugging.

Reply:
Will make the changes after completing the features 👍